suppresses show-more facet links for bucketed range constraints

joemfb opened the pull request:
closes #74

grtjn commented:
I think you can also prevent the show more link upfront, somewhere here: https://github.com/joemfb/ml-search-ng/blob/bucket-show-more/src/ml-search.service.js#L811. Bucketed facets have a distinct type:

```
		"updated": {
			"type": "bucketed",
			"facetValues": [{
				"name": "week ago",
				"count": 2,
				"value": "week ago"
			}, {
				"name": "month ago",
				"count": 4,
				"value": "month ago"
			}, {
				"name": "year ago",
				"count": 6,
				"value": "year ago"
			}, {
				"name": "longer",
				"count": 1,
				"value": "longer"
			}]
		},
```

joemfb commented:
You're right; I had mistakenly thought that bucketed facets showed the type of the underlying index ...

I changed the existing check to an error / promise rejection, and added a check for each facet object. I still need another test for `ml-facets` ...

grtjn commented:
Silly me, was pointing to the complete wrong line..

grtjn commented on src/directives/ml-facets.directive.js:
```diff
@@ -61,6 +61,14 @@
+    $scope.$watch('facets', function(newVal, oldVal) {
+      _.each(newVal, function(facet) {
+        if ( facet.type === 'bucketed' ) {
+          facet.displayingAll = true;
```
Sorry, was pointing to wrong line. I meant to say, you could also toggle this flag when you receive the data in the service, where metadata and such is rewritten..

joemfb replied:
That's a good point, I'll push an update.

joemfb commented:
It's a good call, much simpler this way. I'll add a test confirming it's working as expected.

grtjn commented:
I'll see if I can run a test with this as well, my github-search has bucketed facets, just need to drop in you branch..

grtjn commented:
Whoops, not as easy to test as I thought. The bower.json points to dist/, and can't build ml-search-ng myself..

joemfb commented:
You'll have to build it locally, than copy `dist/` into your bower_components directory.

Something like:

    git fetch upstream bucket-show-more
    git checkout -b bucket-show-more upstream/bucket-show-more

    gulp
    cp dist/* $project/bower_components/ml-search-ng/dist

grtjn commented:
Manually copied the changes, that seemed to work nicely..

grtjn commented:
Oh, I thought I didn't have the build stuff, but I do. Duh.. I have tested manually, I think this looks good..

joemfb commented:
I've added a test for the bucketed facet annotation. Between that and your test, I'm comfortable merging.
